store/download_test.go

```diff
@@ -435,20 +441,102 @@ func (s *downloadSuite) TestUseDeltas(c *C) {
 	}
 
 	for _, scenario := range scenarios {
-		if scenario.exeInCore {
-			osutil.CopyFile("/bin/true", exeInCorePath, 0)
-		} else {
-			os.Remove(exeInCorePath)
-		}
+		var hostCmd, coreCmd *testutil.MockCmd
+
+		comment := Commentf("%#v", scenario)
+
+		// setup the env var for the scenario
 		os.Setenv("SNAPD_USE_DELTAS_EXPERIMENTAL", scenario.env)
 		release.MockOnClassic(scenario.classic)
+
+		// setup binaries for the scenario
+		if scenario.exeInCore {
+			// we need both the xdelta3 command for determining the interpreter
+			// as well as the actual interpreter for executing the basic
+			// "xdelta3 config" command
+			// for the interpreter, since that's how we execute xdelta3, mock
+			// that as a command, but for the xdelta3 command we need the
+			// PT_INTERP symbol set for the binary for elf handling to work,
+			// which won't be the case on a basic shell script
+
+			// TODO: what if the unit tests are run on a system where /bin/true
+			// does not have a PT_INTERP which equals lib64/ld-linux-x86-64.so.2
+			// i.e. lp builders ?
```

Comment:
this will presumably be a real problem that needs to be solved before landing this PR, would be happy to hear suggestions on how to handle this gracefully

Reply:
I think that the unit tests should not call the real `CommandFromSystemSnap()` function, but a mocked version. I see that `overlord/snapstate/backend/fontconfig.go` and `snap/squashfs/squashfs.go` use this function with a level of indirection that allows injecting a mocked version; maybe you should try the same here?

Reply:
this might be easier indeed